src/registers/model_specific.rs

```diff
@@ -445,4 +474,35 @@ mod x86_64 {
             unsafe { msr.write(value.bits()) };
         }
     }
+
+    impl UCet {
+        /// Read the raw IA32_U_CET.
+        #[inline]
+        pub fn read_raw() -> u64 {
+            unsafe { Self::MSR.read() }
+        }
+
+        /// Write the raw IA32_U_CET.
+        #[inline]
+        pub fn write_raw(value: u64) {
+            let mut msr = Self::MSR;
+            unsafe {
+                msr.write(value);
+            }
+        }
+
+        /// Read configurable flags from IA32_U_CET.
+        #[inline]
+        pub fn read_flags() -> UCetFlags {
```

Comment:
IA32_U_CET can also contain a linear address to a legacy code page bitmap (`EB_LEG_BITMAP_BASE`). Perhaps we can return a tuple of the address and the flags [like we do for `Cr3`](https://github.com/rust-osdev/x86_64/blob/33b4c2d394ca93445dc1f77a0a66c57cdc1a7661/src/registers/control.rs#L263).

Reply:
WFM